include/seqan3/core/concepts/core.hpp

```diff
@@ -0,0 +1,285 @@
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+// Author: Rene Rahn <rene.rahn AT fu-berlin.de>
+// ============================================================================
+
+#pragma once
+
+#include <range/v3/range_concepts.hpp>
+#include <range/v3/utility/functional.hpp>
+
+/*!\file core/concepts/core.hpp
+ * \brief Adaptions of core concepts from the Ranges TS.
+ * \ingroup core
+ */
+
+namespace seqan3
+{
+
+//!\name Core Language Concepts
+//!\{
+//!\brief Resolves to `ranges::Same<types...>()` 
+/*!
+ * \sa http://en.cppreference.com/w/cpp/experimental/ranges/iterator/Same
+ */
+template <typename ...ts>
+concept bool same_concept =                         static_cast<bool>(ranges::Same<ts...>());
+
+//!\brief Resolves to `ranges::DerivedFrom<type1, type2>()` 
+/*!
+ * \sa http://en.cppreference.com/w/cpp/experimental/ranges/iterator/DerivedFrom
+ */
+template <typename t, typename u>
+concept bool derived_from_conept =                  static_cast<bool>(ranges::DerivedFrom<t, u>());
+
+//!\brief Resolves to `ranges::ImplicitlyConvertibleTo<type1, type2>()`
+template <typename t, typename u>
+concept bool implicitly_convertible_to_concept =    static_cast<bool>(ranges::ImplicitlyConvertibleTo<t, u>());
+
+//!\brief Resolves to `ranges::ExplicitlyConvertibleTo<type1, type2>()`
+template <typename t, typename u>
+concept bool explicitly_convertible_to_concept =    static_cast<bool>(ranges::ExplicitlyConvertibleTo<t, u>());
+
+//!\brief Resolves to `ranges::ConvertibleTo<type1, type2>()` 
+/*!
```

Comment:
If you have multiple lines, put it all in one block, i.e. start with `/*!` in the first line